packages/ns-api/files/ns.flashstart

```diff
@@ -11,66 +11,118 @@ import sys
+        try:
+            port = int(destination[1])
+            if port < 1 or port > 65535:
+                return False
+        except ValueError:
+            return False
+    return True
+
+
+def __fetch_data():
     return {
         'enabled': e_uci.get('flashstart', 'global', 'enabled', dtype=bool, default=False),
+        'proplus': e_uci.get('flashstart', 'global', 'proplus', dtype=bool, default=False),
         'username': e_uci.get('flashstart', 'global', 'username', default=''),
         'password': e_uci.get('flashstart', 'global', 'password', default=''),
         'zones': [zone for zone in e_uci.get('flashstart', 'global', 'zones', list=True, default=[])
                   if zone != ''],
         'bypass': [bypass for bypass in e_uci.get('flashstart', 'global', 'bypass', list=True, default=[])
-                   if bypass != '']
+                   if bypass != ''],
+        'custom_servers': [item for item in e_uci.get('flashstart', 'global', 'custom_servers', list=True, default=[])
+                           if item != '']
     }
 
 
+e_uci = EUci()
 cmd = sys.argv[1]
 
 if cmd == 'list':
     print(json.dumps({
         'list-zones': {},
         'set-config': {
             'enabled': False,
+            'proplus': False,
             'username': 'str',
             'password': 'str',
             'zones': 'str',
-            'bypass': 'str'
+            'bypass': 'str',
+            'custom_servers': 'str',
         },
-        'get-config': {}
+        'get-config': {},
     }))
 elif cmd == 'call':
     action = sys.argv[2]
-    e_uci = EUci()
     if action == 'set-config':
         try:
-            # decode data
-            data = json.JSONDecoder().decode(sys.stdin.read())
+            data = json.load(sys.stdin)
 
-            # save data
-            e_uci.set('flashstart', 'global', 'enabled', bool(data['enabled']))
-            e_uci.set('flashstart', 'global', 'username', data['username'])
-            e_uci.set('flashstart', 'global', 'password', data['password'])
-            e_uci.set('flashstart', 'global', 'zones', data['zones'])
-            e_uci.set('flashstart', 'global', 'bypass', data['bypass'])
+            if 'enabled' not in data:
+                raise ValidationError('enabled', 'required')
+            if not isinstance(data['enabled'], bool):
+                raise ValidationError('enabled', 'invalid')
+            if data['enabled']:
+                if 'proplus' not in data:
+                    raise ValidationError('proplus', 'required')
+                if not isinstance(data['proplus'], bool):
+                    raise ValidationError('proplus', 'invalid')
+                if 'username' not in data or data['username'] == '':
+                    raise ValidationError('username', 'required')
+                if 'password' not in data or data['password'] == '':
+                    raise ValidationError('password', 'required')
+                if 'zones' not in data:
```

Comment:
This is a little verbose.
Proposal:
- create a loop
- just use data.get(<var>, default=<val>) and get a good default